sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOParallelReadTest.java

```diff
@@ -612,4 +822,195 @@ private void runTestReadFromTableSource(
 
     pipeline.run();
   }
+
+  @Test
+  public void testReadFromQuerySource() throws Exception {
+    runTestReadFromQuerySource(null, defaultCreateSessionRequest, defaultReadRowsRequest);
+  }
+
+  @Test
+  public void testReadFromQuerySourceWithFilter() throws Exception {
```

Comment:
trivial, but should we allow filter and field selection on query source? It is a bit wired.

Reply:
There are customer use cases for this. Let's chat more about this offline.